site/src/pages/release-notes/1.29.0.mdx

```diff
@@ -0,0 +1,257 @@
+  ```
+- You can now configure <type://TlsEngineType> using <type://ClientFactoryBuilder> or <type://ServerBuilder>. #4962
+  ```java
+  ClientFactory
+    .builder()
+    .tlsEngineType(TlsEngineType.OPENSSL) // 👈👈👈
+    .build();
+  ```
+- You can now easily find both dynamic and static decorators that handle a request
+  via <type://ServiceRequestContext.findService(Class)>. #5670
+  ```java
+  ServerBuilder sb = ...
+  sb.decorator(CorsService.builderForAnyOrigin().newDecorator());
+  ...
+  ServiceRequestContext ctx = ...
+  CorsService corsService = ctx.findService(CorsService.class);
+  assert corsService.config().isAnyOriginSupported();
+  ```
+- You can now use the marshaller specified by gRPC `MethodDescriptor` by setting
+  <type://GrpcClientBuilder#useMethodMarshaller(boolean)> and
+  <type://GrpcServiceBuilder#useMethodMarshaller(boolean)>. #5103 #5630
+  ```java
+  GrpcClientBuilder builder = ...
+  builder.useMethodMarshaller(true); // 👈👈👈
+
+  GrpcServiceBuilder builder = ...
+  builder.useMethodMarshaller(true); // 👈👈👈
+  ```
+- You can now programmatically retrieve the server side metrics via <type://ServerMetrics>. #4992 #5627
+  ```java
+  ServerMetrics metrics = serverConfig.serverMetrics();
+  metrics.activeConnections();
+  metrics.pendingRequests();
+  metrics.activeRequests();
+  ```
+- You can now use a <type://CoroutineHttpService> that runs your service in a coroutine scope. #5442 #5603
+  ```kotlin
+  ServerBuilder sb = ...
+  sb.service(
+    "/hello",
+    CoroutineHttpService { ctx, req ->
+      HttpResponse.of("hello world")
+    })
+  ```
+- You can now specify options for an <type://HttpService> via <type://ServiceOptions>. #5071 #5574
+  ```java
+  static final ServiceOptions SERVICE_OPTIONS =
+    ServiceOptions
+      .builder()
+      .requestTimeoutMillis(5000)
+      .maxRequestLength(1024)
+      .requestAutoAbortDelayMillis(1000)
+      .build();
+
+  HttpService httpService = new HttpService() {
+    ...
+    @Override
+    public ServiceOptions options() {
+      return SERVICE_OPTIONS;
+    }
+  };
+
+  // Or use annotation for an annotated service.
+  class MyService {
+      @ServiceOption(requestTimeoutMillis = 5000, maxRequestLength = 1024)
+      @Get("/hello")
+      public HttpResponse hello() {
+          return HttpResponse.of("Hello!");
+      }
+  }
+  ```
+- You can now inject a custom attribute from a <type://ServiceRequestContext> to an annotated service
+  using <type://@Attribute> annotation. #5514 #5547
+  ```java
+  class MyAttributes {
+    public static final AttributeKey<String> USERNAME =
+      AttributeKey.valueOf(MyAttributes.class, "USERNAME");
+  }
+
+  class MyAnnotatedService {
+
+    @Get("/hello")
+    public HttpResponse hello(
+      @Attribute(prefix = MyAttributes.class, value = "USERNAME") // 👈👈👈
+      String username) { ... }
+  }
+  ```
+- You can now specify a graceful shutdown timeout for a <type://ClientFactory>. #5470 #5489
```

Comment:
Technically, it is a graceful timeout for HTTP/2 connection but not `ClientFactory`.
```suggestion
- You can now specify a graceful shutdown timeout for an HTTP/2 connection 
  in <type://ClientFactory>. #5470 #5489
```